niworkflows/utils/spaces.py

```diff
@@ -508,17 +508,55 @@ def insert(self, index, value, error=True):
         elif error is True:
             raise ValueError('space "%s" already in spaces.' % str(value))
 
-    def get_std_spaces(self, only_names=True, dim=(2, 3)):
-        """Return only standard spaces."""
-        names = []
-        std_spaces = []
-        for s in self._spaces:
-            if s.standard and s.dim in dim and s.fullname not in names:
-                names.append(s.fullname)
-                std_spaces.append(s)
-        if only_names:
-            return names
-        return std_spaces
+    def get_std_spaces(self, only_names=True, dim=(2, 3), cached=False):
```

Comment:
I've been reflecting on this a bit more.

Why not returning a new `SpatialReferences` object when calling ``spaces.cache``? That way, all these methods would operate on the cached version without additional code.